### PR TITLE
NASL documentation rework

### PR DESCRIPTION
**Pull request by Kraemii**
**What**:
A new documentation with the goal to have a complete documentation about the OpenVAS project, which is available for everyone. This PR only contains the tooling for building a HTML documentation of the OpenVAS-scanner, also containing a NASL documentation. It is also possible to generate man-pages for the built-in NASL functions. Currently only a few NASL functions are documented, but the missing ones will be added over time.

SC-614

**Why**:
A more centered documentation, which is available for everyone.

**How**:
In addition to the Doxygen documentation, it is now also possible to generate a HTML documentation, which contains useful information about the project. To build the HTML documentation call `make manual`. This will generate the manual within the docs folder.
To generate the man pages for the NASL built-in functions call `make nasl-man`

Note that pandoc is required to built those.

**Checklist**:

- [ ] Tests
- [ ] PR merge commit message adjusted

### CODE REVIEW

**Comment by y0urself** [comment]
We should add the documentation to https://greenbone.github.io/docs/latest/index.html afterwards?

**Comment by ArnoStiefvater** [comment]
> We should add the documentation to https://greenbone.github.io/docs/latest/index.html afterwards?

Yes, but it will take some time (months) until the new documentation is finished. This is just the beginning of it. I would only put it into https://greenbone.github.io/docs/latest/index.html once it is finished.

File: doc/openvas/README.md [hunk 1]
```diff
@@ -0,0 +1,4 @@
+# OpenVAS Documentation
+
+To generate thr Table of Content markdown file run `./toc.sh`
```
Comment (jjnicola) [inline doc/openvas/README.md]: Small typo

Comment (Kraemii) [reply doc/openvas/README.md]: Something was a broken about this PR, as this file did not even existed anymore in the PR branch. I updated the branch and hopefully it is solved

**Review by jjnicola — CHANGES_REQUESTED** [review]
In the INSTALL.md,  the make options must be updated

- Must be added:
  - make doxygen
  - make doxygen-xml
  - make manual
- Remove doc-full

File: doc/man.sh [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    for entry in "$search_dir"/*
+    do 
+        # In case of folder iterate through it
+        if [[ -d $entry ]]; then
+            search_dir="$entry"
+            recursive_functions
+        # Else make an entry for the file
+        elif [[ -f $entry ]]; then
+            filename="$(basename -- $entry)"
+            if [ $filename != index.md && $filename ]; then
```
Comment (jjnicola) [inline doc/man.sh:29]: I got the following error when I run `make nasl-man`
```
Scanning dependencies of target nasl-man
./man.sh: line 29: [: missing `]'
./man.sh: line 29: [: missing `]'
./man.sh: line 29: [: missing `]'
./man.sh: line 29: [: missing `]'
./man.sh: line 29: [: missing `]'
./man.sh: line 29: [: missing `]'
./man.sh: line 29: [: missing `]'
Built target nasl-man
```

**Review by jjnicola — CHANGES_REQUESTED** [review]

File: doc/man.sh (hunk 2 above)
Comment (Kraemii) [reply doc/man.sh]: This is fixed now

**Review by ArnoStiefvater — APPROVED** [review]

**Review by jjnicola — APPROVED** [review]

**Pull request merged**